balena/types/models.py

```diff
@@ -157,6 +158,10 @@ class TypeApplication(TypedDict):
     can_use__application_as_host: ReverseNavigationResource[ApplicationHostedOnApplication]
 
 
+class TypeApplicationWithDeviceServiceDetails(TypeApplication):
+    owns__device: List["TypeDeviceWithServices"]  # type: ignore
```

Comment:
FWIW this is necessary as by default Python type hints won't allow overriding properties on `TypedDict`s